dp_check/dp_check.go

```diff
@@ -142,6 +227,35 @@ func main() {
 	if syscall.Getuid() != 0 {
 		infoLog.Println("Not running as root, some checks may fail.")
 	}
+
+	// Check if dp_check is running on a VM
+	runCheck("Running on GCE VM", func() error {
+		if !isRunningOnGCP() {
+			return fmt.Errorf(`dp_check is not running on a GCE VM`)
+		}
+		return nil
+	})
+
+	// Check connection to metadata server
+	runCheck("HTTP connectivity to metadata server", func() error {
+		client := &http.Client{}
+		metadataServerUrl := "http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ipv6s"
+		req, err := http.NewRequest("GET", metadataServerUrl, nil)
```

Comment:
Can we save the response is successful and compare it to the address observed locally?